### PR TITLE
Add flyway feature

### PR DESCRIPTION
**Pull request by asgrigoryan**
Add flyway feature
https://github.com/nareknaltakyan1/coreapi-auth/issues/1

### CODE REVIEW

File: src/main/resources/db/migration/V01__create_users_table.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+CREATE TABLE _user (
+    id BIGSERIAL PRIMARY KEY,
+    firstname VARCHAR(255) NOT NULL,
+    lastname VARCHAR(255) NOT NULL,
+    email VARCHAR(255) NOT NULL UNIQUE,
+    password VARCHAR(255) NOT NULL,
+    role VARCHAR(20) NOT NULL CHECK (role IN ('USER', 'ADMIN'))
+);
```
Comment (nareknaltakyan1) [inline src/main/resources/db/migration/V01__create_users_table.sql:13]: please add indexes

File: src/main/java/com/nnaltakyan/core/auth/domain/auth/service/AuthenticationService.java [hunk 2]
```diff
@@ -26,8 +26,14 @@ public class AuthenticationService
...
-		var user = User.builder().firstName(registerRequest.getFirstName()).lastName(registerRequest.getLastName()).email(registerRequest.getEmail())
-			.password(passwordEncoder.encode(registerRequest.getPassword())).role(Role.USER).build();
+		var user = User.builder()
+				.firstName(registerRequest.getFirstName())
+				.lastName(registerRequest.getLastName())
+				.email(registerRequest.getEmail())
+				.password(passwordEncoder.encode(registerRequest.getPassword())).role(Role.USER)
+				.created(registerRequest.getCreated())
+				.updated(registerRequest.getUpdated())
+				.deleted(registerRequest.getDeleted()).build();
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/auth/service/AuthenticationService.java]: deleted must be null

File: src/main/java/com/nnaltakyan/core/auth/domain/user/model/User.java [hunk 3]
```diff
@@ -17,7 +18,13 @@
 @NoArgsConstructor
 @AllArgsConstructor
 @Entity
-@Table(name = "_user")
+@Table(name = "_user", indexes = {
+		@Index(name = "idx_user_email", columnList = "email"),
+		@Index(name = "idx_user_created", columnList = "created") },
+		uniqueConstraints = {
+		@UniqueConstraint(name = "nn_email_unique", columnNames = "email")
```
Comment (nareknaltakyan1) [inline src/main/java/com/nnaltakyan/core/auth/domain/user/model/User.java:25]: please use correct name "idx_user_created"

File: src/main/resources/db/migration/V01__create_users_table.sql [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    firstname VARCHAR(255),
+    lastname VARCHAR(255),
+    email VARCHAR(255),
+    password VARCHAR(255),
+    role VARCHAR(20),
+    created TIMESTAMP,
+    deleted TIMESTAMP,
+    updated TIMESTAMP,
+    PRIMARY KEY(id),
+    CONSTRAINT nn_email_not_null CHECK (email IS NOT NULL),
```
Comment (nareknaltakyan1) [inline src/main/resources/db/migration/V01__create_users_table.sql]: please do not write not-null constraint separately

**Review by nareknaltakyan1 — CHANGES_REQUESTED** [review]

File: src/main/resources/db/migration/V01__create_users_table.sql (hunk 4 above)
Comment (asgrigoryan) [reply src/main/resources/db/migration/V01__create_users_table.sql]: Done

File: src/main/java/com/nnaltakyan/core/auth/domain/auth/service/AuthenticationService.java (hunk 2 above)
Comment (asgrigoryan) [reply src/main/java/com/nnaltakyan/core/auth/domain/auth/service/AuthenticationService.java]: Done

File: src/main/resources/db/migration/V01__create_users_table.sql [hunk 5]
```diff
@@ -0,0 +1,16 @@
+CREATE TABLE "_user" (
+    id BIGSERIAL,
```
Comment (nareknaltakyan1) [inline src/main/resources/db/migration/V01__create_users_table.sql]: please add not null

File: src/main/resources/db/migration/V01__create_users_table.sql [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    firstname VARCHAR(255) NOT NULL,
+    lastname VARCHAR(255) NOT NULL,
+    email VARCHAR(255) NOT NULL,
+    password VARCHAR(255) NOT NULL,
+    role VARCHAR(20) NOT NULL,
+    created TIMESTAMP NOT NULL,
+    deleted TIMESTAMP NOT NULL,
+    updated TIMESTAMP NOT NULL,
+    PRIMARY KEY(id),
+    CONSTRAINT nn_email_unique UNIQUE (email)
```
Comment (nareknaltakyan1) [inline src/main/resources/db/migration/V01__create_users_table.sql]: why nn? does it mean "not null"?

**Pull request merged**